integ-test/src/test/java/org/opensearch/sql/sql/DateTimeFormatsIT.java

```diff
@@ -120,6 +122,31 @@ public void testNumericFormats() {
         rows("1970-01-02 03:55:00", "1970-01-01 00:01:40.5"));
   }
 
+  @Test
+  @SneakyThrows
+  public void testDateNanosWithFormats() {
+    String query = String.format("SELECT hour_minute_second_OR_t_time"
+        + " FROM %s", TEST_INDEX_DATE_FORMATS);
+    JSONObject result = executeQuery(query);
+    verifySchema(result,
+        schema("hour_minute_second_OR_t_time", null, "time"));
+    verifyDataRows(result,
+        rows("09:07:42"),
+        rows("09:07:42"));
+  }
+
+  @Test
+  @SneakyThrows
+  public void testDateNanosWithNanos() {
+    String query = String.format("SELECT date_nanos_value"
+        + " FROM %s", TEST_INDEX_DATATYPE_NONNUMERIC);
+    JSONObject result = executeQuery(query);
+    verifySchema(result,
+        schema("date_nanos_value", null, "timestamp"));
+    verifyDataRows(result,
+        rows("2019-03-24 01:34:46.123456789"));
+  }
```

Comment:
We need to ensure that we can convert date_nanoes to date type.

Reply:
Are there any additional requirements on date_nanos?  Should that type work with any other existing function, or does it first need to be cast to `date` type?

Reply:
No need for a cast, it is converted to the same internal types as `date` - `OpenSearchDateType`.